# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1295519983 18000
#      Thu Jan 20 05:39:43 2011 -0500
# Node ID 6006ca07410d3a92a3b47f2984a1513bf27f7361
# Parent  e306683a7e973c78eb6544cf785f68767eb257dc
another data member initialization fix

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,14 @@
+2011-01-20  John W. Eaton  <jwe@octave.org>
+
+	* oct-obj.h (octave_value_list::octave_value_list): Initialize
+	all data members in initialization list.  From Pascal Dupuis
+	<Pascal.Dupuis@worldonline.be>.
+
 2011-01-20  John W. Eaton  <jwe@octave.org>
 
 	* ls-mat5.cc (read_mat5_binary_element):
 	Improve diagnositc if uncompress fails.
 
 	* load-path.cc (strip_trailing_separators): New static function.
 	(load_path::do_add, load_path::do_remove): Call it on directory arg.
 
diff --git a/src/oct-obj.h b/src/oct-obj.h
--- a/src/oct-obj.h
+++ b/src/oct-obj.h
@@ -36,32 +36,32 @@ along with Octave; see the file COPYING.
 
 class
 OCTINTERP_API
 octave_value_list
 {
 public:
 
   octave_value_list (void)
-    : data () { }
+    : data (), names () { }
 
   explicit octave_value_list (octave_idx_type n)
-    : data (dim_vector (1, n)) { }
+    : data (dim_vector (1, n)), names () { }
 
   octave_value_list (octave_idx_type n, const octave_value& val)
-    : data (dim_vector (1, n), val) { }
+    : data (dim_vector (1, n), val), names () { }
 
   octave_value_list (const octave_value& tc)
-    : data (dim_vector (1, 1), tc) { }
+    : data (dim_vector (1, 1), tc), names () { }
 
   octave_value_list (const Array<octave_value>& d)
-    : data (d.as_row ()) { }
+    : data (d.as_row ()), names () { }
 
   octave_value_list (const Cell& tc)
-    : data (tc.as_row ()) { }
+    : data (tc.as_row ()), names () { }
 
   octave_value_list (const octave_value_list& obj)
     : data (obj.data), names (obj.names) { }
 
   // Concatenation constructor.
   octave_value_list (const std::list<octave_value_list>&);
 
   ~octave_value_list (void) { }
